sphinx_gallery/gen_gallery.py

```diff
@@ -29,6 +29,7 @@
 DEFAULT_GALLERY_CONF = {
     'filename_pattern': re.escape(os.sep) + 'plot',
     'examples_dirs': os.path.join('..', 'examples'),
+    'examples_dirs_order': None,
```

Comment:
`None` feels very strange to me, like giving the option to be completely random. It also hides the type we expect in this field. I would set as default the empty dictionary.